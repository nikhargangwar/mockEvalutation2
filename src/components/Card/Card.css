.card{
    height: 200px;
    width: 200px;
    border: 1px solid black;
    border-radius: 20px;
    margin-bottom: 15px;
    
}
.card-image-wrapper{
    height: 78%;
    width: 80%;
    margin: 0 auto;
    display: flex;
    align-items: center;
}

.card-image{
    height: 80%;
    width: 80%;
    border-radius: 1000px;
    border: 1px solid black;
    overflow: hidden;
    margin: 0 auto;
    
}
.card-image img{
        width: 100%;
        height: auto;
        /* background-size: cover;
        background-repeat: no-repeat;
        background-position: center;
          object-fit: contain;

        flex-shrink: 0; */
        object-fit: contain;

}
.card-details{
    display: flex;
}
.card-text{
    flex: 1;
    margin-left: 20px;
}
.card-heading{
    font-size: 20px;
    font-weight: 200;
}

.card-subheading{
    font-size:10px;
    font-weight: 900;
}
.liked-icon {
    margin-right: 10px;
    position: relative;
}

.liked-icon img{
    height: 25px;
    width: 25px;
   margin-top: 8px;
}

.liked-icon h3{
    position:absolute;
    margin-top: 10px;
    margin-left: 8px;
    font-size: 15px;
    z-index: 10;
    color: white;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-weight: 300;
}