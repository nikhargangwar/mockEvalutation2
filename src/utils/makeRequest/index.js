import axios from 'axios';
import { ERROR_ROUTE } from '../../constants/routesPaths';
import { BACKEND_URL } from '../../constants/apiEndPoints';

const makeRequest = async ({ url, method }, dynamicConfig, navigate) => {
  try {
    const response = await axios({
      baseURL: BACKEND_URL,
      url,
      method,
      ...dynamicConfig,
    });
    return response.data;
  } catch (e) {
    if (navigate) {
      const errorStatus = e.response?.status;

      if (errorStatus) {
        navigate(`${ERROR_ROUTE}/${errorStatus}`);
      } else {
        navigate(ERROR_ROUTE);
      }
    }
    return {};
  }
};

export default makeRequest;
