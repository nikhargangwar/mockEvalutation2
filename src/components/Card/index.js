/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React, { useEffect, useState } from 'react';
import './Card.css';
import GET_LIKES_DATA from '../../constants/likesEndPoint.json';
import makeRequest from '../../utils/makeRequest';
import greyHeartIcon from '../../assets/heart-gray.svg';
import redHeartIcon from '../../assets/heart-red.svg';

function Card({ songdata }) {
  const [likeCount, setLikeCount] = useState(0);
  const [isLiked, setIsLiked] = useState(false);

  useEffect(() => {
    makeRequest({ ...GET_LIKES_DATA, url: `${GET_LIKES_DATA.url}/${songdata.id}/likes` })
      .then((response) => {
        setLikeCount(response.data.count);
        setIsLiked(response.data.like);
      });
  }, []);
  return (
    <div className="card">
      <div className="card-image-wrapper">
        <div className="card-image">
          <img src={songdata.imageUrl} alt="title" />
        </div>
      </div>

      <div className="card-details">
        <div className="card-text">
          <div className="card-heading">
            {songdata.name}
          </div>
          <div className="card-subheading">
            {songdata.artist.name}
          </div>
        </div>
        <div className="liked-icon">
          <h3>{likeCount}</h3>
          {isLiked ? (<img src={redHeartIcon} alt="hearticon" />)
            : (<img src={greyHeartIcon} alt="hearticon" />)}

        </div>
      </div>
    </div>
  );
}

export default Card;
