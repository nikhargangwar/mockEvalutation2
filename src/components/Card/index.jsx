/* eslint-disable */
/* eslint-disable*/
import React, { useEffect, useState } from 'react';
import './Card.css';
import GET_LIKES_DATA from '../../constants/likesEndPoint.json';
import makeRequest from '../../utils/makeRequest';
import greyHeartIcon from '../../assets/heart-gray.svg';
import redHeartIcon from '../../assets/heart-red.svg';
import { GET_SONG_DATA, LIKE_SONG } from '../../constants/apiEndPoints';

function Card({ id, songdata }) {
  const [likeCount, setLikeCount] = useState(0);
  const [isLiked, setIsLiked] = useState(false);
  // const [cardClass, setCardClass] = useState(false);

  let cardClass;
  if (id % 2 === 0) {
    cardClass='grey-bakground';
    // setCardClass('grey-bakground');
  } else {
    cardClass='dark-bakground';
    // setCardClass('dark-background');
  }

  const clickHandler = () => {
    if (isLiked) {
      makeRequest(
        LIKE_SONG(songdata.id),
        {
          headers: { authorization: 'Bearer QWlzaHdhcnlhIE4=' },
          data: { like: false },
        },
      );
      setIsLiked(!isLiked);
      setLikeCount(likeCount - 1);
    } else {
      makeRequest(
        LIKE_SONG(songdata.id),
        {
          headers: { authorization: 'Bearer QWlzaHdhcnlhIE4=' },
          data: { like: true },
        },
      );
      setIsLiked(!isLiked);
      setLikeCount(likeCount + 1);
    }
  };

  useEffect(() => {
    makeRequest(GET_SONG_DATA(songdata.id), {
      headers: { authorization: 'Bearer QWlzaHdhcnlhIE4=' },
    })
      .then((response) => {
        setLikeCount(response.data.count);
        setIsLiked(response.data.like);
      });
  }, []);

  
  return (
    <div className={cardClass}>
      <div className="card-image-wrapper">
        <div className="card-image">
          <img src={songdata.imageUrl} alt="title" />
        </div>
      </div>

      <div className="card-details">
        <div className="card-text">
          <div className="card-heading">
            {songdata.name}
          </div>
          <div className="card-subheading">
            {songdata.artist.name}
          </div>
        </div>
        <div className="liked-icon">
          <h3>
            <button type="button" onClick={clickHandler}>
              {likeCount}
            </button>
          </h3>
          {isLiked ? (<img src={redHeartIcon} alt="hearticon" />)
            : (<img src={greyHeartIcon} alt="hearticon" />)}

        </div>
      </div>
    </div>
  );
}

export default Card;
