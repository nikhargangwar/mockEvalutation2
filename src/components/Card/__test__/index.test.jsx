import {
  render, screen, fireEvent, waitFor, getByRole,
} from '@testing-library/react';
import React from 'react';
import Card from '..';
import makeRequest from '../../../utils/makeRequest';

import mockSingleSongData from '../../../mocks/singleSongData';

jest.mock('../../../utils/makeRequest');

describe('Card component', () => {
  it('should render when mounted', () => {
    const { container } = render(<Card
      id={1}
      songdata={mockSingleSongData}
    />);
    expect(container).toMatchSnapshot();
  });

  //   it('should render the card', async () => {
  //     makeRequest.mockResolvedValue({
  //       data: {
  //         count: 15,
  //         like: true,
  //       },
  //     });
  //     render(<Card
  //       id={1}
  //       songdata={mockSingleSongData}
  //     />);
  //     expect(screen.getByText('Lost')).toBeTruthy();
  //     await waitFor(() => {
  //       expect(screen.getByText('15')).toBeTruthy();
  //     });
  //   });

  it('should render the card with like false', () => {
    makeRequest.mockResolvedValueOnce({
      data: {
        count: 15,
        like: false,
      },
    });
    makeRequest.mockResolvedValueOnce({
      data: {
        count: 16,
        like: true,
      },
    });
    const { container } = render(<Card
      id={1}
      songdata={mockSingleSongData}
    />);
    waitFor(() => {
      expect(screen.queryByText('15')).toBeTruthy();
    });

    const likeButton = container.getBy('testButton');
    fireEvent.click(likeButton);

    waitFor(() => {
      expect(screen.queryByText('16')).toBeTruthy();
    });
  });

  //   it('should increment thhe like count by 1 when not liked', () => {
  //     makeRequest.mockResolvedValueOnce({
  //       data: {
  //         count: 15,
  //         like: false,
  //       },
  //     });
  //     makeRequest.mockResolvedValueOnce({
  //       data: {
  //         count: 16,
  //         like: true,
  //       },
  //     });

//     render(<Card
//       id={1}
//       songdata={mockSingleSongData}
//     />);
//     makeRequest.mockResolvedValue();
//     const likeButton = getByRole('button');
//     fireEvent.click(likeButton);
//     expect(screen.getByText(10));
//   });
});
