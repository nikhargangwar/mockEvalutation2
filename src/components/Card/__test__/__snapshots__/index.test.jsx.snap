// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render when mounted 1`] = `
<div>
  <div
    class="dark-bakground"
    data-testid="testCard"
  >
    <div
      class="card-image-wrapper"
    >
      <div
        class="card-image"
      >
        <img
          alt="title"
          src="https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631"
        />
      </div>
    </div>
    <div
      class="card-details"
    >
      <div
        class="card-text"
      >
        <div
          class="card-heading"
        >
          Lost
        </div>
        <div
          class="card-subheading"
        >
          Maroon 5
        </div>
      </div>
      <div
        class="liked-icon"
      >
        <h3>
          <button
            type="button"
          >
            0
          </button>
        </h3>
        <img
          alt="hearticon"
          src="heart-gray.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
